### ISSUE TITLE
Lua error in 2.0.16 on WoW 10.0.2 "attempt to call method 'IsUnit' (a nil value)"

### ISSUE THREAD

**Issue by Arrhae** [wow-rp-addons/XRP#39]
Seems to happen occasionally when mousing over a player who has an RP profile. Full error from BugSack
```
Interface/AddOns/XRP/Interface/Tooltip/Tooltip.lua:479: attempt to call method 'IsUnit' (a nil value)
[string "=[C]"]: in function `xpcall'
[string "@Interface/AddOns/XRP/Backend/Events.lua"]:36: in function `RunEvent'
[string "@Interface/AddOns/XRP/Backend/MSP.lua"]:142: in function <Interface/AddOns/XRP/Backend/MSP.lua:137>
[string "=[C]"]: in function `xpcall'
[string "@Interface/AddOns/XRP/Libraries/LibMSP/LibMSP.lua"]:175: in function <Interface/AddOns/XRP/Libraries/LibMSP/LibMSP.lua:173>
[string "@Interface/AddOns/XRP/Libraries/LibMSP/LibMSP.lua"]:551: in function <Interface/AddOns/XRP/Libraries/LibMSP/LibMSP.lua:520>
[string "@Interface/AddOns/XRP/Libraries/LibMSP/LibMSP.lua"]:575: in function <Interface/AddOns/XRP/Libraries/LibMSP/LibMSP.lua:563>
[string "=[C]"]: in function `xpcall'
[string "@Interface/AddOns/XRP/Libraries/Chomp/Internal.lua"]:236: in function <Interface/AddOns/XRP/Libraries/Chomp/Internal.lua:116>
[string "@Interface/AddOns/XRP/Libraries/Chomp/Internal.lua"]:633: in function <Interface/AddOns/XRP/Libraries/Chomp/Internal.lua:629>
```

**Comment by Wyr3d** [comment]
Same here, altough this LUA error did not pop up right away

Message: Interface/AddOns/XRP/Interface/Tooltip/Tooltip.lua:479: attempt to call method 'IsUnit' (a nil value)
Time: Thu Nov 17 21:21:59 2022
Count: 1
Stack: Interface/AddOns/XRP/Interface/Tooltip/Tooltip.lua:479: attempt to call method 'IsUnit' (a nil value)
[string "@Interface/AddOns/XRP/Interface/Tooltip/Tooltip.lua"]:479: in function <Interface/AddOns/XRP/Interface/Tooltip/Tooltip.lua:471>
[string "=[C]"]: in function `xpcall'
[string "@Interface/AddOns/XRP/Backend/Events.lua"]:36: in function `RunEvent'
[string "@Interface/AddOns/XRP/Backend/MSP.lua"]:142: in function <Interface/AddOns/XRP/Backend/MSP.lua:137>
[string "=[C]"]: in function `xpcall'
[string "@Interface/AddOns/XRP/Libraries/LibMSP/LibMSP.lua"]:175: in function <Interface/AddOns/XRP/Libraries/LibMSP/LibMSP.lua:173>
[string "@Interface/AddOns/XRP/Libraries/LibMSP/LibMSP.lua"]:551: in function <Interface/AddOns/XRP/Libraries/LibMSP/LibMSP.lua:520>
[string "@Interface/AddOns/XRP/Libraries/LibMSP/LibMSP.lua"]:575: in function <Interface/AddOns/XRP/Libraries/LibMSP/LibMSP.lua:563>
[string "=[C]"]: in function `xpcall'
[string "@Interface/AddOns/XRP/Libraries/Chomp/Internal.lua"]:236: in function <Interface/AddOns/XRP/Libraries/Chomp/Internal.lua:116>
[string "@Interface/AddOns/XRP/Libraries/Chomp/Internal.lua"]:633: in function <Interface/AddOns/XRP/Libraries/Chomp/Internal.lua:629>

Locals: <none>

**Comment by Soulsinger-Dreamshadow** [comment]
Not to butt in, but I know this can be fatal if it throws the error at the wrong time.
You can shut it up by changing line 479 from

`if replace and not GameTooltip:IsUnit("mouseover") then`
  to
`if replace and not UnitExists("mouseover") then`

It's not a perfect solution, but it'll circumvent the change to how tooltips are handled with 10.0.2 and preserve the "biography window"'s ability to fade back out.

**Comment by Wyr3d** [comment]
The other issue I had on my warlock may have been a varient of this. It does looks similar, although the numbers were different. haven't seen it since though

**Comment by Solanya** [comment]
Should be fixed with 2.0.17

**Comment by Arrhae** [comment]
I'm not seeing the error anymore, thank you!
